### PR TITLE
Skip database already exists errors

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
When in single-binary mode, data catalog and admin both attempt to initialize the database at the same time.  Error check when getting an error back from the db create call to skip the error if the other goroutine created it in the meantime.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Spin up a local db, pause it in the function so that `createPostgresDbIfNotExists` will not return early, create the database manually, resume and inspect the error returned.  Testing was done locally with a postgres container.

### CODE REVIEW

**Review by katrogan — DISMISSED** [review]
If we're going to be replicating some of this in datacatalog it might be worthwhile consolidating common code in https://github.com/flyteorg/flytestdlib/tree/master/database - but can be left as a todo

File: pkg/repositories/database.go [hunk 1]
```diff
@@ -139,21 +140,24 @@ func createPostgresDbIfNotExists(ctx context.Context, gormConfig *gorm.Config, p
 	result := gormDb.Exec(createDBStatement)
 
 	if result.Error != nil {
-		return nil, result.Error
+		if !isPgErrorWithCode(result.Error, pqDbAlreadyExistsCode) {
+			logger.Warningf(ctx, "Got DB already exists for [%v], skipping...", pgConfig.DbName)
```
Comment (katrogan) [inline pkg/repositories/database.go:144]: nit: this log statement is wrong, right?

Comment (wild-endeavor) [reply pkg/repositories/database.go:144]: is it? why?

Comment (katrogan) [reply pkg/repositories/database.go:144]: because you're checking that it's not pqDbAlreadyExistsCode

**Review by katrogan — APPROVED** [review]

File: pkg/repositories/database.go (hunk 1 above)
Comment (wild-endeavor) [reply pkg/repositories/database.go]: 🤦
